packages/harmony/src/components/button/Button/Button.tsx

```diff
@@ -59,37 +28,228 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
     } = props
     const { _isHovered, _isPressed, isLoading } = baseProps
     const isDisabled = disabled || isLoading
+    const {
+      type,
+      color: themeColors,
+      cornerRadius,
+      shadows,
+      spacing,
+      typography
+    } = useTheme()
+
+    // Size Styles
+    const smallStyles: CSSObject = {
+      gap: spacing.unit1,
+      height: spacing.unit8,
+      paddingInline: spacing.unit3,
+      fontSize: typography.size.s,
+      fontWeight: typography.weight.bold,
+      lineHeight: spacing.unit4,
+      textTransform: 'capitalize'
+    }
+    const smallIconStyles: CSSObject = {
+      zIndex: 1,
+      width: spacing.unit4,
+      height: spacing.unit4
+    }
+
+    const defaultStyles: CSSObject = {
+      gap: spacing.unit2,
+      height: spacing.unit12,
+      paddingInline: spacing.unit6,
+      fontSize: typography.size.l,
+      fontWeight: typography.weight.bold,
+      // TODO: See if this can be done with theme vars
+      lineHeight: 'calc(4.5 * var(--harmony-unit))',
+      textTransform: 'capitalize'
+    }
+    const defaultIconStyles: CSSObject = {
+      zIndex: 1,
+      width: spacing.unit5,
+      height: spacing.unit5
+    }
+
+    const largeStyles: CSSObject = {
+      gap: spacing.unit2,
+      height: spacing.unit16,
+      paddingInline: spacing.unit6,
+      fontSize: typography.size.xl,
+      fontWeight: typography.weight.bold,
+      lineHeight: spacing.unit4,
+      letterSpacing: '0.25px',
+      textTransform: 'uppercase'
+    }
+    const largeIconStyles: CSSObject = {
+      zIndex: 1,
+      width: spacing.unit6,
+      height: spacing.unit6
+    }
+
+    // Variant Styles
+    const primaryHoverStyles: CSSObject = {
+      '--overlay-color': '255, 255, 255',
+      '--overlay-opacity': 0.2,
+      boxShadow: shadows.mid
+    }
+    const primaryActiveStyles: CSSObject = {
+      '--overlay-color': '0, 0, 0',
```

Comment:
Is there a way to make this work without using css variables? I guess it's somewhat clever, but could just use a JavaScript variable right?